src/query/config/src/outer_v0.rs

```diff
@@ -1217,3 +1228,100 @@ impl Debug for MetaConfig {
+pub struct IDMConfig {
+    users: Vec<UserConfig>,
+}
+
+impl TryInto<InnerIDMConfig> for IDMConfig {
+    type Error = ErrorCode;
+
+    fn try_into(self) -> Result<InnerIDMConfig> {
+        let mut users = HashMap::new();
+        for c in self.users.into_iter() {
+            users.insert(c.name.clone(), c.auth.try_into()?);
+        }
+        Ok(InnerIDMConfig { users })
+    }
+}
+
+impl From<InnerIDMConfig> for IDMConfig {
+    fn from(inner: InnerIDMConfig) -> Self {
+        Self {
+            users: inner
+                .users
+                .into_iter()
+                .map(|(name, auth)| UserConfig {
+                    name,
+                    auth: auth.into(),
+                })
+                .collect(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct UserConfig {
+    pub name: String,
+    #[serde(flatten)]
+    pub auth: UserAuthConfig,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct UserAuthConfig {
+    auth_type: String,
+    auth_string: Option<String>,
+}
+
+fn check_no_auth_string(auth_string: Option<String>, auth_info: AuthInfo) -> Result<AuthInfo> {
+    match auth_string {
+        Some(s) if !s.is_empty() => Err(ErrorCode::InvalidConfig(format!(
+            "should not set auth_string for auth_type {}",
+            auth_info.get_type().to_str()
+        ))),
+        _ => Ok(auth_info),
+    }
+}
+impl TryInto<AuthInfo> for UserAuthConfig {
+    type Error = ErrorCode;
+
+    fn try_into(self) -> Result<AuthInfo> {
+        let auth_type = AuthType::from_str(&self.auth_type)?;
+        match auth_type {
+            AuthType::NoPassword => check_no_auth_string(self.auth_string, AuthInfo::None),
+            AuthType::JWT => check_no_auth_string(self.auth_string, AuthInfo::JWT),
+            AuthType::Sha256Password | AuthType::DoubleSha1Password => {
+                let password_type = auth_type.get_password_type().expect("must success");
+                match self.auth_string {
+                    None => Err(ErrorCode::InvalidConfig("must set auth_string")),
+                    Some(s) => {
+                        let p = hex::decode(s).map_err(|e| {
+                            ErrorCode::InvalidConfig(format!("password is not hex: {e:?}"))
+                        })?;
+                        Ok(AuthInfo::Password {
+                            hash_value: p,
+                            hash_method: password_type,
+                        })
+                    }
+                }
+            }
+        }
+    }
+}
+
+impl From<AuthInfo> for UserAuthConfig {
+    fn from(inner: AuthInfo) -> Self {
+        let auth_type = inner.get_type().to_str().to_owned();
+        let auth_string = inner.get_auth_string();
+        let auth_string = if auth_string.is_empty() {
+            None
+        } else {
+            Some(hex::encode(auth_string))
```

Comment:
why do we need `hex::encode` ?  If `AuthType`  is plain, it should not be encoded.